### PR TITLE
Edit goal

### PR DESCRIPTION
**Pull request by ChrisCruze**
You can now edit goal and delete goal from the Goal Dashboard page

### CODE REVIEW

**Comment by vivaromarf** [comment]
Added "added setTimeout(refetch,2000)" to fix bug

File: Components/Pages/Goals.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ const Goals = ({ navigation }) => {
   const { updateGoal } = useGoalUpdate();
   const { removeGoal } = useGoalDelete();
   refetch();
+setTimeout(refetch,2000)
```
Comment (anandangalig) [inline Components/Pages/Goals.js:34]: these two really should be replaced with useEffect:

- pass in refetch() for the callback (1st param)
- pass in [] to fetch once on render (2nd param) - same as componendDidMount()
- pass in [goals] to fetch only when goals variable changes (2nd param) - same as componendDidUpdate()

File: Components/Pages/editGoal.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import { TextField } from "../Atoms/Fields";
+import { Dropdown } from "react-native-material-dropdown";
+import { useGoalUpdate, useGoalsPull } from "../../API";
+
+const editGoal = ({ navigation }) => {
+  const { goals, refetch } = useGoalsPull();
+  const { updateGoal } = useGoalUpdate();
+
+  const _id = navigation.getParam("_id");
+  refetch();
```
Comment (anandangalig) [inline Components/Pages/editGoal.js:25]: useEffect hook can be used here too?

Comment (ChrisCruze) [reply Components/Pages/editGoal.js:25]: great suggestion - anywhere refetch is, we'll put in useEffect

**Comment by ChrisCruze** [comment]
> goals

Yea - I totally hear ya on that

**Comment by ChrisCruze** [comment]
there's actually one additional feature I'd like to add into this pull request, which is editing/deleting the timestamps under a goal (e.g. if you accidentally pushed, you can delete that push). @anandangalig - mind if I add this quick feature in as well since its related to edit-goal. I'm asking since you already reviewed code and this would add a few more things

**Pull request merged**